tests/firedrake/adjoint/test_burgers_newton.py

```diff
@@ -189,35 +201,54 @@ def test_burgers_newton(solve_type, checkpointing):
                           ("solve", "Mixed"),
                           ("NLVS", "Mixed"),
                           ])
-def test_checkpointing_validity(solve_type, checkpointing):
-    """Compare forward and backward results with and without checkpointing.
-    """
-    mesh, timestep, steps = basics()
-    V = FunctionSpace(mesh, "CG", 2)
+def test_checkpointing_validity(solve_type, checkpointing, basics):
+    """Compare forward and backward results with and without checkpointing."""
+    mesh, timestep, steps = basics
     # Without checkpointing
-    tape = get_working_tape()
-    tape.progress_bar = ProgressBar
-    x, = SpatialCoordinate(mesh)
-    ic = project(sin(2.*pi*x), V)
-
-    val0 = J(ic, solve_type, timestep, steps, V)
+    V, ic, nu = setup_test(mesh)
+    val0 = J(ic, nu, solve_type, timestep, steps, V)
     Jhat = ReducedFunctional(val0, Control(ic))
     dJ0 = Jhat.derivative()
+    tape = get_working_tape()
     tape.clear_tape()
 
     # With checkpointing
     tape.progress_bar = ProgressBar
     if checkpointing == "Revolve":
         tape.enable_checkpointing(Revolve(steps, steps//3))
-    if checkpointing == "Mixed":
+    elif checkpointing == "Mixed":
         enable_disk_checkpointing()
         tape.enable_checkpointing(MixedCheckpointSchedule(steps, steps//3, storage=StorageType.DISK))
         mesh = checkpointable_mesh(mesh)
-    V = FunctionSpace(mesh, "CG", 2)
-    x, = SpatialCoordinate(mesh)
-    ic = project(sin(2.*pi*x), V)
-    val1 = J(ic, solve_type, timestep, steps, V)
+
+    V, ic, nu = setup_test(mesh)
+    # Reinitialize function space and initial condition
+    val1 = J(ic, nu, solve_type, timestep, steps, V)
     Jhat = ReducedFunctional(val1, Control(ic))
+
     assert len(tape.timesteps) == steps
     assert np.allclose(val0, val1)
     assert np.allclose(dJ0.dat.data_ro[:], Jhat.derivative().dat.data_ro[:])
+
+
+@pytest.mark.skipcomplex
+@pytest.mark.parametrize("nu_time_dependent", [True, False])
+def test_global_deps(nu_time_dependent, basics):
+    """Test the global dependencies."""
+    mesh, timestep, steps = basics
+    tape = get_working_tape()
+    tape.enable_checkpointing(Revolve(steps, steps//3))
+    V, ic, nu = setup_test(mesh)
+    val0 = J(ic, nu, "NLVS", timestep, steps, V, nu_time_dependent=nu_time_dependent)
+    Jhat = ReducedFunctional(val0, Control(ic))
+
+    if nu_time_dependent:
```

Comment:
Can there be a comment explaining why this is the expected behaviour? It's not obvious to me.

Reply:
done

Reply:
That genuinely really helps! Thanks!